persistence/commons/src/main/java/org/apache/causeway/persistence/commons/integration/repository/RepositoryServiceDefault.java

```diff
@@ -120,7 +147,9 @@ public <T> T persist(final T domainObject) {
     @Override
     public <T> T persistAndFlush(final T object) {
         persist(object);
-        transactionService.flushTransaction();
+        if (!threadLocal.get().getOrDefault(object.getClass(), Boolean.FALSE)) {
```

Comment:
I'm wondering if this would work for JPA ... doesn't dynamic weaving create a subclass of the original entity?  
In which case, perhaps we can simplify - is there any reason to turn on bulk mode on a class-by-class basis ... why not simply turn it on "globally" (for the thread, that is)?

Reply:
Good point. The idea was to have more fine grain control over which classes you wanted to persist with or without a flush. But in bulk mode you want to have everything in one transaction. Everything else so have its own transaction or even thread.

Reply:
Fixed